pkg/storage/chunk/client/object_client.go

```diff
@@ -98,11 +91,17 @@ func NewClient(store ObjectClient, encoder KeyEncoder, schema config.SchemaConfi
 }
 
 func NewClientWithMaxParallel(store ObjectClient, encoder KeyEncoder, maxParallel int, schema config.SchemaConfig) Client {
+
 	return &client{
 		store:               store,
 		keyEncoder:          encoder,
 		getChunkMaxParallel: maxParallel,
 		schema:              schema,
+		bufferPool: sync.Pool{
+			New: func() interface{} {
+				return bytes.NewBuffer(make([]byte, 0, 1000))
```

Comment:
This is a very small nit:

```suggestion
				return bytes.NewBuffer(make([]byte, 0, 1024))
```

Memory pages are allocated usually in 4KiB or 16KiB pages, so we should align with that boundary to avoid an extra allocation.